kafka-connect-s3/src/test/java/io/confluent/connect/s3/integration/S3SinkConnectorNetworkIT.java

```diff
@@ -0,0 +1,333 @@
+  private static final int NUM_RECORDS_PRODUCED = 1000;
+  private static final int TASKS_MAX = 1;
+  private static final List<String> KAFKA_TOPICS = Arrays.asList("topic1");
+  private static final int FLUSH_SIZE = 200;
+  private static SquidProxy squid;
+
+  @Before
+  public void setup() throws IOException {
+    startConnect();
+    createS3RootClient();
+    // create the test bucket
+    createS3Bucket(S3_BUCKET);
+  }
+
+
+  @After
+  public void close() {
+    // delete the test bucket
+    deleteBucket(S3_BUCKET);
+  }
+
+  /**
+   * Success scenario : End to end test
+   * @throws Throwable
+   */
+  @Test
+  @Ignore
+  public void testToAssertConnectorAndDestinationRecords() throws Throwable {
+
+    // create topics in Kafka
+    KAFKA_TOPICS.forEach(topic -> connect.kafka().createTopic(topic, 1));
+    // send records to kafka
+    int totalNoOfRecordsProduced = sendRecordsToKafka();
+
+    Map<String, String> props = getConnectorProps();
+
+    // start a sink connector
+    connect.configureConnector(CONNECTOR_NAME, props);
+    // wait for tasks to spin up
+    int minimumNumTasks = Math.min(KAFKA_TOPICS.size(), TASKS_MAX);
+
+    waitForConnectorToStart(CONNECTOR_NAME, minimumNumTasks);
+    waitForConnectorToCompleteSendingRecords(totalNoOfRecordsProduced, FLUSH_SIZE);
+
+    // assert records
+    assertFileCountInBucket(S3_BUCKET, totalNoOfRecordsProduced / FLUSH_SIZE);
+  }
+
+  /**
+   * Case in which bucket permissions/policies are changed while uploading records
+   * Prerequisite : Access key and Secret access key should be set as environment variables
+   * @throws Exception
+   */
+  @Test
+  @Ignore
+  public void testWithRevokedWritePermissions() throws InterruptedException {
+
+    addReadWritePolicyToBucket(S3_BUCKET);
+    KAFKA_TOPICS.forEach(topic -> connect.kafka().createTopic(topic, 1));
+
+    // send records to kafka
+    int totalNoOfRecordsProduced = sendRecordsToKafka();
+
+    Map<String, String> props = getConnectorProps();
+    props.put("aws.access.key.id", System.getenv("SECONDARY_USER_ACCESS_KEY_ID"));
+    props.put("aws.secret.access.key", System.getenv("SECONDARY_USER_SECRET_ACCESS_KEY"));
+
+    // start a sink connector
+    connect.configureConnector(CONNECTOR_NAME, props);
+    // wait for tasks to spin up
+    int minimumNumTasks = Math.min(KAFKA_TOPICS.size(), TASKS_MAX);
+
+    waitForConnectorToStart(CONNECTOR_NAME, minimumNumTasks);
+    waitForConnectorToCompleteSendingRecords(totalNoOfRecordsProduced, FLUSH_SIZE);
+
+    // revoke read/write permission
+    alterReadWritePolicyOfBucket(S3_BUCKET);
+    /*
+     Intentional sleep added in order for the bucket permission to be altered to read only to
+     come into affect.
+    */
+    Thread.sleep(10000);
+    // produce more records to kafka
+    totalNoOfRecordsProduced += sendRecordsToKafka();
+    int idealObjectCount = totalNoOfRecordsProduced / FLUSH_SIZE;
+    int actualObjectWritten = NUM_RECORDS_PRODUCED / FLUSH_SIZE;
```

Comment:
is this the amount of files expected per call to `sendRecordsToKafka` ? Let's use a more descriptive var name as to what this should be.

Reply:
Yes, I was using the `NUM_RECORD_PRODUCED` as a constant to the number of records to be sent to Kafka each time the `sendRecordsToKafka()` is called. 
I have refactored the code to use `fileCountBeforeRevokingPermission` instead of `actualObjectWritten`  and also initialized it at a suitable place for better code readability.